### PR TITLE
Fix lost subscriptions when authentication tokens expire

### PR DESCRIPTION
**Pull request by benoitvidis**
## What does this PR do ?

This PR fixes the situation where a user would 
1. log in
2. subscribe to some rooms
3. after some time, log in again
4. see his subscriptions lost once his first login expires

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "new-cap": 1,
     "no-caller": 2,
     "no-catch-shadow": 2,
+    "no-console": 2,
```
Comment (benoitvidis) [inline .eslintrc.json]: removed from eslint:recommended

File: lib/api/core/auth/tokenManager.js [hunk 2]
```diff
@@ -172,22 +169,20 @@ class TokenManager {
...
-        expiresAt: newToken.expiresAt,
-        userId: newToken.userId,
-        rooms: this.tokens.array[pos].rooms
-      });
-      this.tokens.remove({idx: oldIndex});
+      this._add(
+        newToken,
+        this.tokens.array[pos].connectionId,
+        [...this.tokens.array[pos].rooms]
+      );
```
Comment (Yoann-Abbes) [inline lib/api/core/auth/tokenManager.js]: (nit)
this should be on the line above according to the new wrapping rules

Comment (benoitvidis) [reply lib/api/core/auth/tokenManager.js]: :+1:

File: lib/api/core/auth/tokenManager.js [hunk 3]
```diff
@@ -62,6 +63,7 @@ class TokenManager {
 
       return a.idx < b.idx ? -1 : 1;
     });
+    this.tokensByConnectedUser = {};
```
Comment (Aschen) [inline lib/api/core/auth/tokenManager.js]: Can't we use a `Map` instead?

Comment (benoitvidis) [reply lib/api/core/auth/tokenManager.js]: :+1:

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "kuzzle",
   "author": "The Kuzzle Team <support@kuzzle.io>",
-  "version": "1.9.1",
+  "version": "1.9.2",
```
Comment (scottinet) [inline package.json]: The package-lock.json file must also be updated.

Comment (benoitvidis) [reply package.json]: :+1:

**Pull request merged**